cmd/otel-allocator/allocation/allocator_test.go

```diff
@@ -0,0 +1,191 @@
+	for _, i := range s.collectors {
+		assert.True(t, i.NumTargets == 25)
+	}
+
+}
+
+func TestCollectorBalanceWhenAddingAndRemovingAtRandom(t *testing.T) {
+
+	// prepare allocator with 3 collectors and 'random' amount of targets
+	s := NewAllocator()
+	cols := []string{"col-1", "col-2", "col-3"}
+	s.SetCollectors(cols)
+
+	tar := []string{"targ:1001", "targ:1002", "targ:1003", "targ:1004", "targ:1005", "targ:1006",
+		"targ:1011", "targ:1012", "targ:1013", "targ:1014", "targ:1015", "targ:1016",
+		"targ:1021", "targ:1022", "targ:1023", "targ:1024", "targ:1025", "targ:1026"}
+	var tarL []TargetItem
+	for _, i := range tar {
+		tarL = append(tarL, TargetItem{JobName: "sample-name", TargetURL: i, Label: model.LabelSet{}})
+	}
+	s.SetWaitingTargets(tarL)
+	s.Reallocate()
+
+	// removing targets at 'random'
+	tar = []string{"targ:1002", "targ:1003", "targ:1004", "targ:1006",
+		"targ:1011", "targ:1012", "targ:1013", "targ:1014", "targ:1016",
+		"targ:1023", "targ:1024", "targ:1025", "targ:1026"}
+	tarL = []TargetItem{}
+	for _, i := range tar {
+		tarL = append(tarL, TargetItem{JobName: "sample-name", TargetURL: i, Label: model.LabelSet{}})
+	}
+	s.SetWaitingTargets(tarL)
+	s.Reallocate()
+
+	// adding targets at 'random'
+	tar = []string{"targ:1002", "targ:1003", "targ:1004", "targ:1006",
+		"targ:1011", "targ:1012", "targ:1001", "targ:1014", "targ:1016",
+		"targ:1023", "targ:1024", "targ:1025", "targ:1126", "targ:1227"}
+	tarL = []TargetItem{}
+	for _, i := range tar {
+		tarL = append(tarL, TargetItem{JobName: "sample-name", TargetURL: i, Label: model.LabelSet{}})
+	}
+	s.SetWaitingTargets(tarL)
+	s.Reallocate()
+
+	count := len(s.targetItems) / len(s.collectors)
+	percent := float64(len(s.targetItems)) / 6.7
+	upperBound := float64(count) + percent
+	lowerBound := float64(count) - percent
+
+	for _, i := range s.collectors {
+		assert.True(t, float64(i.NumTargets) >= lowerBound && float64(i.NumTargets) <= upperBound)
+	}
+}
+
+func TestCollectorBalanceWhenCollectorsChanged(t *testing.T) {
+	s := NewAllocator()
+	cols := []string{"col-1", "col-2"}
+	s.SetCollectors(cols)
+
+	tar := []string{"targ:1001", "targ:1002", "targ:1003", "targ:1004", "targ:1005", "targ:1006",
+		"targ:1011", "targ:1012", "targ:1013", "targ:1014", "targ:1015", "targ:1016",
+		"targ:1021", "targ:1022", "targ:1023", "targ:1024", "targ:1025", "targ:1026"}
+	var tarL []TargetItem
+	for _, i := range tar {
+		tarL = append(tarL, TargetItem{JobName: "sample-name", TargetURL: i, Label: model.LabelSet{}})
+	}
+	s.SetWaitingTargets(tarL)
+	s.Reallocate()
+
+	// removing targets at 'random'
+	tar = []string{"targ:1002", "targ:1003", "targ:1004", "targ:1006",
+		"targ:1011", "targ:1012", "targ:1013", "targ:1014", "targ:1016",
+		"targ:1023", "targ:1024", "targ:1025", "targ:1026"}
+	tarL = []TargetItem{}
+	for _, i := range tar {
+		tarL = append(tarL, TargetItem{JobName: "sample-name", TargetURL: i, Label: model.LabelSet{}})
+	}
+	s.SetWaitingTargets(tarL)
```

Comment:
Its impossible to tell which collector will have a certain amount of targets since they are being added and removed in the multiple steps. This test is just checking that we go back to a ground up allocation when collectors change. 
Perhaps i should remove the noisy steps and just test for the functionality mentioned